### DISCUSSION TITLE
Best Practices for Sentinel with Multiple Databases

### DISCUSSION THREAD

**Discussion by jkratz55** [redis/go-redis#2373] [Q&A]
I'm using Redis to store cached product data but currently caching it by different floor plans, IE: US, vs International. Because they have the same keys I am keeping them in different Redis databases 0 and 1 respectively. I was using `NewFailoverClusterClient` because I want to route reads to replicas for max throughput and performance. However, everything was written to DB 0. After looking at the code it made sense since `NewFailoverClusterClient` is just an adapter for ClusterClient which doesn't support multiple DBs. So I suppose my question is what are the best practices if I want to use Sentinel and multiple databases? Ideally, I'd like reads to be distributed but it isn't clear to me that `NewFailureClient` doesn't just send all reads/writes to whatever node is currently the master.

**Comment by jkratz55** [comment]
Not sure if the same team maintains this documentation, but it would probably be helpful to call out explicitly that using `NewFailoverClusterClient` has the same behavior as using ClusterClient in that you can only have a single DB.

https://redis.uptrace.dev/guide/go-redis-sentinel.html#redis-server-client

**Comment by monkey92t** [comment]
`FailoverClient` support read replica nodes?
OR.... `FailoverClusterClient` supports DB-1?

**Reply by jkratz55** [reply]
Sorry, let me try to clarify what I'm asking.

If I use the `NewFailoverClient` function, it returns a *redis.Client. When I use that client, is it only communicating with the node that is currently the master? That is to say; if doing read and write operations, do they all go to the current master?

When using `NewFailoverClusterClient` it returns a ClusterClient, and looking at the code, it appears to ignore the DB field in the FailoverOptions struct, which makes sense as cluster mode doesn't support multiple databases.

Ultimately I want to run Redis deployed in Sentinel mode, but I'd like the client to send read-only operations to all replicas but also support using multiple databases.
